llvm/lib/CodeGen/SelectionDAG/LegalizeVectorTypes.cpp

```diff
@@ -4749,6 +4755,14 @@ void DAGTypeLegalizer::WidenVectorResult(SDNode *N, unsigned ResNo) {
   case ISD::VP_FSHR:
     Res = WidenVecRes_Ternary(N);
     break;
+  case ISD::FSINCOS: {
+    if (!unrollExpandedOp())
+      Res = WidenVecRes_FSINCOS(N);
+    for (unsigned ResNum = 0; ResNum < N->getNumValues(); ResNum++)
+      SetWidenedVector(SDValue(N, ResNum), Res.getValue(ResNum));
+    Res = SDValue();
```

Comment:
Yes, it should unroll of possible (which is does for `<3 x float>`). There are some cases where it does not unroll right now, but probably should, but that's a general limitation of the "should unroll" check.

Reply:
> Is it worth creating a WidenVecRes_UnaryOpWithTwoResults for this? (it seems FREXPR is missing in this switch statement)

It may be, but I think creating something to handle both `FFREXP` and `FSINCOS` may take a little more care. `FSINCOS` is easy as both results are the same type (so have the same legalization action), but that might be something that needs handling differently for `FFREXP` (which returns an int and a float).

Reply:
Done this now, and enabled widening for `FREXPR` (with two AArch64 tests).